### PR TITLE
MAINT: Split up .github/workflows to match main

### PR DESCRIPTION
**Pull request by charris**
This is a backport of #24493, #24291, and #24625. The purpose of this is to ease future
backports that expect these files.

- CI: move some jobs in `build_test.yml` to Meson

- CI: split `build_test.yml` into three GHA jobs files

  Also documents better what is being run. See gh-24410 for the
  overall restructuring plan for GitHub Actions CI.

- CI: merge `linux_meson.yml` into `linux_blas.yml`

- TST: disable mypy tests in test suite unless an environment variable is set

  These tests are super slow, and they're effectively always passing in CI.
  Running them on all "full" test suite runs is too expensive. Note that
  SciPy has an XSLOW mark, NumPy does not. So use an env var for now.

- CI: add new GHA CI jobs to run MyPy across OS/Python flavors

### CODE REVIEW

File: .github/workflows/linux_simd.yml [hunk 1]
```diff
@@ -26,51 +53,53 @@ permissions:
...
         submodules: recursive
         fetch-depth: 0
     - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
       with:
-        python-version: ${{ env.PYTHON_VERSION }}
-    - uses: ./.github/meson_actions
+        python-version: ${{ matrix.python-version }}
+    - uses: ./.github/actions
 
+||||||| parent of 1a81895ae6 (CI: split `build_test.yml` into three GHA jobs files):.github/workflows/build_test.yml
```
Comment (rgommers) [inline .github/workflows/linux_simd.yml:91]: left-over merge snippet here

File: .github/workflows/linux_simd.yml [hunk 2]
```diff
@@ -79,17 +108,18 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - uses: ./.github/actions
 
+=======
+>>>>>>> 1a81895ae6 (CI: split `build_test.yml` into three GHA jobs files):.github/workflows/linux_simd.yml
```
Comment (rgommers) [inline .github/workflows/linux_simd.yml:112]: and here

File: .github/workflows/linux_simd.yml [hunk 3]
```diff
@@ -178,9 +201,10 @@ jobs:
         fetch-depth: 0
     - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
       with:
-        python-version: ${{ env.PYTHON_VERSION }}
+        python-version: '3.10'
     - uses: ./.github/meson_actions
 
+<<<<<<< HEAD:.github/workflows/build_test.yml
```
Comment (rgommers) [inline .github/workflows/linux_simd.yml:207]: and here

File: .github/workflows/linux_simd.yml [hunk 4]
```diff
@@ -269,60 +308,182 @@ jobs:
...
+    - name: Install test dependencies
+      run: |
+        pip install -r test_requirements.txt
+    - name: Run test suite
+      run: |
+        cd doc
+        pytest --pyargs numpy -m "not slow"
+
+
+||||||| parent of 1a81895ae6 (CI: split `build_test.yml` into three GHA jobs files):.github/workflows/build_test.yml
```
Comment (rgommers) [inline .github/workflows/linux_simd.yml:346]: and here

File: .github/workflows/linux_simd.yml [hunk 5]
```diff
@@ -269,60 +308,182 @@ jobs:
...
+      run: |
+        pip install -r test_requirements.txt
+    - name: Run test suite
+      run: |
+        cd doc
+        pytest --pyargs numpy -m "not slow"
+
+
+=======
+>>>>>>> 1a81895ae6 (CI: split `build_test.yml` into three GHA jobs files):.github/workflows/linux_simd.yml
```
Comment (rgommers) [inline .github/workflows/linux_simd.yml:486]: and here

**Review by rgommers — COMMENTED** [review]
Seems useful indeed, thanks Chuck. LGTM modulo a few stray merge conflict markers.

File: .github/workflows/linux_simd.yml (hunk 4 above)
Comment (charris) [reply .github/workflows/linux_simd.yml:346]: Thanks @rgommers I think those bits come from a different backport I was working on, one of the reasons for this one. Fixed now.

**Comment by charris** [comment]
The Linux SIMD test is failing for Python 3.12, I currently have it commented out.

**Comment by rgommers** [comment]
> The Linux SIMD test is failing for Python 3.12, I currently have it commented out.

Perhaps easier to debug separately? This is green and looks safe to merge. I'm happy to have a look at the failure after this gets merged.

**Comment by charris** [comment]
> This is green and looks safe to merge.

Let's do that. Trying to wring information from the meson log isn't easy.

**Pull request merged**